src/LabLog.Domain/Entities/Room.cs

```diff
@@ -20,6 +21,11 @@ public Room()
 
         public static Room Create(string name, Action<ILabEvent> eventHandler)
         {
+            if (name == "")
+            {
+                LabException ex = new LabException("Room name can't be blank");
+                throw ex;
+            }

```

Comment:
How can we check if a name has already been used? Would need to get all the RoomCreatedEvents and RoomNameChangedEvents, and build up a state of the name for every room? How?

Reply:
Hmmmm. This is a good point, and might indicate a flaw in the model. I'll have a think

Reply:
I think either it means the Aggregate Root is wrong, and it should be School (which I think is not true), or that this constraint isn't really a function of the model. Your school could definitely name two rooms the same thing, they just shouldn't.
But you could help by maintaining a list of all the currently used room names, like an aggregate read model, and validating against that. Not as an absolutely certain check, but as a helper to data input.

Reply:
If the Room is the aggregate root, then storing it in the same place as the things inside it doesn't quite make sense. Perhaps a store of Room id and names in a separate table in the database, separate from the events. The events would then relate to those Rooms. Then the rooms could be queried separately from the events, and solve the problem. This does sound like its heading towards the idea of maintaining state in the database as well as the events.

Reply:
That wouldn't solve the problem of race conditions though. The whole point was using the write model for validation, not the read model. And the write model isn't aware of the database context.

Reply:
How about a universal Aggregate?
Computer <- Room <- School <- District <- State <- Country <- Continent <- Planet <- Galaxy... you see where I'm going with this :P

Reply:
To make things more complicated, I want to be able to access a room based off its name rather than id. /Admin/room/Room19